test/functional/blockchain.py

```diff
@@ -44,6 +45,38 @@ def run_test(self):
         self._test_stopatheight()
         assert self.nodes[0].verifychain(4, 0)
 
+    def _test_getblockchaininfo(self):
+        keys = [
+            'bestblockhash',
+            'bip9_softforks',
+            'blocks',
+            'chain',
+            'chainwork',
+            'difficulty',
+            'headers',
+            'mediantime',
+            'pruned',
+            'softforks',
+            'verificationprogress',
+        ]
+        res = self.nodes[0].getblockchaininfo()
+        # should have exact fields
+        assert_equal(sorted(res.keys()), keys)
+
+        # restart node with pruning enabled
+        self.stop_node(0)
+        self.start_node(0, extra_args=["-prune=1"])
+
+        res = self.nodes[0].getblockchaininfo()
+        # should also have pruneheight field
+        assert_equal(sorted(res.keys()), sorted(['pruneheight'] + keys))
+        # should be greater or equal to 0
+        assert res['pruneheight'] >= 0
+
+        # restore node
+        self.stop_node(0)
+        self.start_node(0, self.extra_args[0])
```

Comment:
I think you can just use `self.start_node(0)`. The default will use `self.extra_args[0]` for its args.